core/core_modules/disable.js

```diff
@@ -1,24 +1,124 @@
+            threads[event.thread_id].msgIndexEnable = 2;
+            threads[event.thread_id].msgIndexDisable = 3;
+            counter = 0;
+            return true;
+        }
+    }
+    return threads[event.thread_id].isThreadDisabled;
 };
 
-exports.help = function(commandPrefix) {
-	return [[commandPrefix + 'disable','Disables the platform', 'Stops the platform from responding to messages it receives.\nTo renable send the disable command again']];
+exports.run = function (api, event) {
+    // Command /disable /counter <value> (Stateless)
+    if (event.arguments[0] === api.commandPrefix + commands[0] &&
+        event.arguments_body.startsWith(api.commandPrefix + commands[1])) {
+        parseInput(event.arguments_body.substring((api.commandPrefix + commands[1]).length, event.arguments_body.length),
+            function (input) {
+                threads[event.thread_id].counterLimit = input;
+                api.sendMessage(messages[5] + ' ' + event.sender_name, event.thread_id);
+            });
+        return false;
+
+        // Command /disable /timer <seconds> (Stateless)
+    } else if (event.arguments[0] === api.commandPrefix + commands[0] &&
+        event.arguments_body.startsWith(api.commandPrefix + commands[2])) {
+        parseInput(event.arguments_body.substring((api.commandPrefix + commands[2]).length, event.arguments_body.length),
+            function (input) {
+                setTimeout(function () {
+                    threads[event.thread_id].isThreadDisabled = !threads[event.thread_id].isThreadDisabled;
+                }, seconds * 1000); // Converting seconds to milliseconds
+                api.sendMessage(messages[8] + ' ' + event.sender_name, event.thread_id);
+            });
+        return false;
+
+        // Command /disable /default (Stateless)
+    } else if (event.arguments[0] === api.commandPrefix + commands[0] &&
+        event.arguments_body === api.commandPrefix + commands[4]) {
+        threads[event.thread_id].counterLimit = 3;
+        api.sendMessage(messages[6] + ' ' + event.sender_name, event.thread_id);
+        return false;
+
+
+        // Main Branch (State-dependent)
+    } else if (event.arguments[0] === api.commandPrefix + commands[0] ||
+        threads[event.thread_id].possibleSpam) {
+        if (threads[event.thread_id].isThreadDisabled) {
+            api.sendMessage(messages[threads[event.thread_id].msgIndexEnable] + ' ' + event.sender_name, event.thread_id);
+        } else {
+            api.sendMessage(messages[threads[event.thread_id].msgIndexDisable], event.thread_id);
+        }
+        threads[event.thread_id].isThreadDisabled = !threads[event.thread_id].isThreadDisabled;
+        threads[event.thread_id].possibleSpam = false;
+
+        // Error handling
+    } else {
+        api.sendMessage(messages[7] + ' ' + event.sender_name, event.thread_id);
+    }
+    return false;
+};
+
+function parseInput(api, event, input, callback) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['parseInput' is defined but never used.](https://www.codacy.com/app/theprojectshop/Kassy/file/2562621006/issues/source?bid=3282987&fileBranchId=3282995#l110)